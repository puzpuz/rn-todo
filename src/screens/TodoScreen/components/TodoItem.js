import React from "react";
import {
  Text,
  TouchableOpacity,
  TouchableWithoutFeedback,
  View,
} from "react-native";
import styles from "./styles/todoItemStyles";

const TodoItem = ({
  item = undefined,
  onRemoveTodo = () => {},
  onSetUpdateItem = () => {},
}) => {
  return (
    <TouchableWithoutFeedback onPress={() => onSetUpdateItem(item)}>
      <View style={styles.todoItemContainer}>
        <Text style={styles.text}>{item?.value}</Text>
        <TouchableOpacity>
          <Text style={styles.removeButton} onPress={() => onRemoveTodo(item)}>
            Remove
          </Text>
        </TouchableOpacity>
      </View>
    </TouchableWithoutFeedback>
  );
};

export default TodoItem;
