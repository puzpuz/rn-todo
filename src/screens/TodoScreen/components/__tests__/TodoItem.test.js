import renderer from "react-test-renderer";
import TodoItem from "../TodoItem";
it("renders correctly", () => {
  const tree = renderer.create(<TodoItem />).toJSON();
  expect(tree).toMatchSnapshot();
});
