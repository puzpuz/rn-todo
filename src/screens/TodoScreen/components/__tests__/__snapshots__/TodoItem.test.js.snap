// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "blue",
      "borderRadius": 40,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginBottom": 8,
      "paddingHorizontal": 16,
      "paddingVertical": 16,
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "color": "#fff",
        "fontSize": 16,
      }
    }
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Text
      onPress={[Function]}
      style={
        {
          "color": "#a0a0a0",
          "fontSize": 16,
        }
      }
    >
      Remove
    </Text>
  </View>
</View>
`;
